rust/src/main.rs

```diff
@@ -11,19 +11,48 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-use std::io;
 
 #[macro_use]
 extern crate log;
 extern crate chrono;
 extern crate fern;
 
+extern crate dirs;
+
 extern crate xi_core_lib;
 extern crate xi_rpc;
 
+use std::io;
+use std::fs;
+use std::path::PathBuf;
+
 use xi_core_lib::XiCore;
 use xi_rpc::RpcLoop;
 
+const XI_DIRECTORY: &'static str = "xi-core";
+const XI_LOG_FILE: &'static str = "xi-core.log";
+
+fn get_logging_directory() -> Option<PathBuf> {
+    if let Some(mut log_dir) = dirs::data_local_dir() {
+        log_dir.push(XI_DIRECTORY);
+        Some(log_dir)
+    } else {
+        eprintln!("[WARNING] The dir library was not able to find a directory for this platform");
+        None
+    }
+}
+
+fn path_for_log_file<P: AsRef<str>>(filename: P) -> Option<PathBuf> {
```

Comment:
although it really shouldn't matter, you should be able to have `AsRef<Path>` instead of `AsRef<str>`. `String` and `&str` both implement `AsRef<Path>`, so you can still call this with string types, but you can also call it with a `Path` or `PathBuf`, and the intent is clearer.

Reply:
I looked into using `PathBuf` instead of `str` but:
> the trait `std::convert::AsRef<std::path::PathBuf>` is not implemented for `str`

Using `Path` worked fine though.